sql/src/main/java/io/crate/execution/dml/upsert/ReturnValueGen.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.execution.dml.upsert;
+
+import io.crate.data.Input;
+import io.crate.execution.engine.collect.CollectExpression;
+import io.crate.expression.InputFactory;
+import io.crate.expression.reference.Doc;
+import io.crate.expression.reference.DocRefResolver;
+import io.crate.expression.symbol.Symbol;
+import io.crate.metadata.Functions;
+import io.crate.metadata.TransactionContext;
+import io.crate.metadata.doc.DocTableInfo;
+
+import javax.annotation.Nullable;
+import java.util.List;
+
+
+final class ReturnValueGen {
+
+    private final List<CollectExpression<Doc, ?>> expressions;
+    private final List<Input<?>> inputs;
+
+    ReturnValueGen(Functions functions, TransactionContext txnCtx, DocTableInfo table, Symbol[] returnValues) {
+        InputFactory.Context<CollectExpression<Doc, ?>> cntx = new InputFactory(functions).ctxForRefs(
+            txnCtx, new DocRefResolver(table.partitionedBy()));
+        cntx.add(List.of(returnValues));
```

Comment:
`returnValues` where nullable everywhere, it is kind of strange that somewhere on the way here it was handled and now it we do not check for null, thus, i'd say that it is much easier to work with empty lists..

Reply:
It is not annotated with `@Nullable`, therefore it is expected to be not null and it is checked before https://github.com/crate/crate/pull/9434/files#diff-6bac7c4ae891a36186ba15afc9483b94R146

Reply:
and this one is not annotated with @nullable, but can be null https://github.com/crate/crate/pull/9434/files#diff-fda033de9b86cee5957d9e1c1b8ed58dR95

imho, it is kind of dangerous

Reply:
Thanks for catching the last one.